### PR TITLE
Fix some deprecation warnings in docs

### PR DESCRIPTION
**Pull request by thomas88**
- Using plain javascript arrays instead of Immutable.List for mention suggestions (docs, stories)
- Using new github-buttons api (fixes deprecation warning in docs)
- Fixing typo
- Removing unused css reference

### CODE REVIEW

File: docs/template.js [hunk 1]
```diff
@@ -22,7 +22,6 @@ const head = `<meta charset="utf-8"/>
 <link rel="stylesheet" href="/css/normalize.css"/>
 <link rel="stylesheet" href="/css/base.css"/>
 <link rel="stylesheet" href="/css/Draft.css"/>
-<link rel="stylesheet" href="/app.css"/>
```
Comment (juliankrispel) [inline docs/template.js]: why is this removed???

File: docs/client/components/pages/Home/index.js [hunk 2]
```diff
@@ -127,7 +127,7 @@ export default class App extends Component {
             <div className={styles.logoText}>DraftJS Plugins</div>
             <p className={styles.tagline}>High quality plugins with great UX</p>
             <div className={styles.githubWrapper} ref={(element) => { this.githubWrapper = element; }}>
-              <GithubButton user="draft-js-plugins" repo="draft-js-plugins" size="mega" />
+              <GithubButton user="draft-js-plugins" repo="draft-js-plugins" size="large" />
```
Comment (juliankrispel) [inline docs/client/components/pages/Home/index.js]: why change this?

File: docs/client/components/pages/Mention/CustomComponentMentionEditor/mentions.js [hunk 3]
```diff
@@ -31,6 +29,6 @@ const mentions = fromJS([
     title: 'HeathIT hacker and researcher',
     avatar: 'https://pbs.twimg.com/profile_images/688487813025640448/E6O6I011_400x400.png',
   },
-]);
```
Comment (juliankrispel) [inline docs/client/components/pages/Mention/CustomComponentMentionEditor/mentions.js]: thx :+1:

File: docs/client/components/shared/GithubButton/index.js [hunk 4]
```diff
@@ -19,19 +19,17 @@ export default class GithubStarButton extends Component {
...
     // Note: all of the attributes including the className 'github-button' are required
     return (
       <a
         ref={(element) => { this.githubButton = element; }}
         className="github-button"
-        href="https://github.com/draft-js-plugins/draft-js-plugins"
-        data-style={size}
-        data-count-href="/draft-js-plugins/draft-js-plugins/stargazers"
-        data-count-api="/repos/draft-js-plugins/draft-js-plugins#stargazers_count"
-        data-count-aria-label="# stargazers on GitHub"
```
Comment (juliankrispel) [inline docs/client/components/shared/GithubButton/index.js]: hold on what's going on here? :)

File: docs/client/components/shared/SocialBar/index.js [hunk 5]
```diff
@@ -10,7 +10,7 @@ export default class SocialBar extends Component {
     return (
       <div className={styles.root}>
         <div className={styles.githubButtonWrapper}>
-          <GithubButton user="draft-js-plugins" repo="draft-js-plugins" size="mega" />
+          <GithubButton user="draft-js-plugins" repo="draft-js-plugins" size="large" />
```
Comment (juliankrispel) [inline docs/client/components/shared/SocialBar/index.js]: ok why?

File: docs/client/components/shared/GithubButton/index.js (hunk 4 above)
Comment (thomas88) [reply docs/client/components/shared/GithubButton/index.js]: I was getting these deprecation warnings in console:

screen shot 2017-09-29 at 11 32 32

Thus I've updated our code to use the new api according to the [github-buttons docs](https://github.com/ntkme/github-buttons#readme). Also we were setting `user` and `repo` props, but not using them in the `GithubButtons` component.

File: docs/client/components/shared/SocialBar/index.js (hunk 5 above)
Comment (thomas88) [reply docs/client/components/shared/SocialBar/index.js]: According to [github-buttons docs](https://github.com/ntkme/github-buttons#readme) the size parameter can be "None by default or large". Large renders the same as on the current page.

screen shot 2017-09-29 at 11 41 43

File: docs/client/components/pages/Home/index.js (hunk 2 above)
Comment (thomas88) [reply docs/client/components/pages/Home/index.js]: See [other comment](https://github.com/draft-js-plugins/draft-js-plugins/pull/896#discussion_r141828927) :)

Comment (juliankrispel) [reply docs/client/components/pages/Home/index.js]: cool :+1:

File: docs/template.js (hunk 1 above)
Comment (juliankrispel) [reply docs/template.js]: @thomas88 all looks good except this??

Comment (thomas88) [reply docs/template.js]: Great that you look over this diligently, I've made a mistake here.

The line was removed, as I was seeing following warning message in console and found that the file wasn't existing.

screen shot 2017-09-29 at 11 45 01

However I didn't understand that template.js is also used in production, where app.css is the built version of the css.

It would be great to find a way to remove the warning in development. For now I'm adding the file to the template again.

Comment (juliankrispel) [reply docs/template.js]: @thomas88 awesome thanks. I'm working on new [documentation btw, using gitbook](https://github.com/draft-js-plugins/draft-js-plugins/pull/899) so all of this will soon go away, and it'll be easier to write and read docs!
